src/pgapp_worker.erl

```diff
@@ -44,7 +44,14 @@ squery(Sql, Timeout) ->
 squery(PoolName, Sql, Timeout) ->
     middle_man_transaction(PoolName,
                            fun (W) ->
-                                   gen_server:call(W, {squery, Sql}, Timeout)
+                                   try
+                                       gen_server:call(W, {squery, Sql},
+                                                       Timeout)
+                                   catch
+                                       exit:E ->
+                                           exit(W, kill),
+                                           exit(E)
```

Comment:
In general, probably it would be better to keep original stacktrace by using `erlang:raise(exit, E, get_stacktrace())`